src/cookbook/packaging-components-npm.md

```diff
@@ -0,0 +1,197 @@
+
+```json
+{
+  "name": "my-component",
+  "version": "1.2.3",
+  "main": "dist/my-component.umd.js",
+  "module": "dist/my-component.esm.js",
+  "unpkg": "dist/my-component.min.js",
+  "browser": {
+    "./sfc": "src/my-component.vue"
+  },
+  "scripts": {
+    "build": "npm run build:umd & npm run build:es & npm run build:unpkg",
+    "build:umd": "rollup --config build/rollup.config.js --format umd --file dist/my-component.umd.js",
+    "build:es": "rollup --config build/rollup.config.js --format es --file dist/my-component.esm.js",
+    "build:unpkg": "rollup --config build/rollup.config.js --format iife --file dist/my-component.min.js"
+  },
+  "devDependencies": {
+    "rollup": "^1.17.0",
+    "@rollup/plugin-buble": "^0.21.3",
+    "@rollup/plugin-commonjs": "^11.1.0",
+    "rollup-plugin-vue": "^5.0.1",
+    "vue": "^2.6.10",
+    "vue-template-compiler": "^2.6.10"
+    ...
+  },
+  ...
+}
+```
+
+Our changes to `package.json` are complete. Next, we need a small wrapper to export the actual SFC, plus a minimal Rollup configuration, and we're set!
+
+### What does my packaged component look like?
+
+Depending on how your component is being used, it needs to be exposed as either a [CommonJS/UMD](https://medium.freecodecamp.org/javascript-modules-a-beginner-s-guide-783f7d7a5fcc#c33a) javascript module, an [ES6 javascript](https://medium.freecodecamp.org/javascript-modules-a-beginner-s-guide-783f7d7a5fcc#4f5e) module so it's immediately available to the page. This is accomplished by a `wrapper.js` file which handles the module export and auto-install. That wrapper, in its entirety, looks like this:
+
+```js
+// Import vue component
+import component from './my-component.vue'
+
+// Declare install function executed by Vue.use()
+export function install(Vue) {
+  if (install.installed) return
+  install.installed = true
+  Vue.component('MyComponent', component)
+}
+
+// Create module definition for Vue.use()
+const plugin = {
+  install
+}
+
+// Auto-install when vue is found (eg. in browser via <script> tag)
+let GlobalVue = null
+if (typeof window !== 'undefined') {
+  GlobalVue = window.Vue
+} else if (typeof global !== 'undefined') {
+  GlobalVue = global.Vue
+}
+if (GlobalVue) {
+  GlobalVue.use(plugin)
+}
+
+// To allow use as module (npm/webpack/etc.) export component
+export default component
+```
+
+Notice the first line directly imports your SFC, and the last line exports it unchanged. As indicated by the comments in the rest of the code, the wrapper provides an `install` function for Vue, then attempts to detect Vue and automatically install the component. With 90% of the work done, it's time to sprint to the finish!
+
+### How do I configure the Rollup build?
+
+With the package.json `scripts` section ready and the SFC wrapper in place, all that is left is to ensure Rollup is properly configured. Fortunately, this can be done with a small 16 line rollup.config.js file:
```

Comment:
```suggestion
With the `package.json` `scripts` section ready and the SFC wrapper in place, all that is left is to ensure Rollup is properly configured. Fortunately, this can be done with a small, 18 line rollup.config.js file:
```